auto_submit/lib/service/secrets.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:googleapis/secretmanager/v1.dart';
+
+/// Access secrets for Google Cloud projects.
+///
+/// See also:
+///   * https://cloud.google.com/secret-manager/docs
+abstract class SecretManager {
+  const SecretManager();
+
+  Future<String> get(String key);
+
+  void put(String key, [String? value]);
+}
+
+class CloudSecretManager implements SecretManager {
+  CloudSecretManager();
+
+  final String projectId = Platform.environment['APPLICATION_ID'] ?? 'flutter-dashboard';
```

Comment:
Nit: simply use `flutter-dashboard` as this is not local run and these secrets are in this project only.

Reply:
I want to upstream this into `package:gcloud`, and I'm leaving this as a breadcrumb to sort out how to find this (googleapis surfaces it somewhere). SG?